### ISSUE TITLE
Build fails on Ubuntu 20.04

### ISSUE THREAD

**Issue by brendandburns** [deislabs/akri#41]
I ran:

```sh
./build/setup.sh
cargo build
```

And I see:
```sh
bburns@helios:~/src/akri$ RUST_BACKTRACE=1 cargo build
   Compiling hyper v0.13.7
   Compiling async-std v1.6.2
   Compiling agent v0.0.36 (/home/bburns/src/akri/agent)
   Compiling udev-video-broker v0.0.36 (/home/bburns/src/akri/samples/brokers/udev-video-broker)
error: failed to run custom build command for `udev-video-broker v0.0.36 (/home/bburns/src/akri/samples/brokers/udev-video-broker)`

Caused by:
  process didn't exit successfully: `/home/bburns/src/akri/target/debug/build/udev-video-broker-6ad08c5c40d22b43/build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /home/bburns/.cargo/registry/src/github.com-1ecc6299db9ec823/tonic-build-0.1.1/src/lib.rs:219:19
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /usr/src/rustc-1.43.0/vendor/backtrace/src/backtrace/libunwind.rs:86
   1: backtrace::backtrace::trace_unsynchronized
             at /usr/src/rustc-1.43.0/vendor/backtrace/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:78
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1063
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1426
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:204
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:224
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:474
  11: rust_begin_unwind
             at src/libstd/panicking.rs:378
  12: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
  13: core::result::unwrap_failed
             at src/libcore/result.rs:1222
  14: core::result::Result<T,E>::unwrap
             at /usr/src/rustc-1.43.0/src/libcore/result.rs:1003
  15: tonic_build::fmt
             at /home/bburns/.cargo/registry/src/github.com-1ecc6299db9ec823/tonic-build-0.1.1/src/lib.rs:219
  16: tonic_build::Builder::compile
             at /home/bburns/.cargo/registry/src/github.com-1ecc6299db9ec823/tonic-build-0.1.1/src/lib.rs:172
  17: build_script_build::main
             at samples/brokers/udev-video-broker/build.rs:2
  18: std::rt::lang_start::{{closure}}
             at /usr/src/rustc-1.43.0/src/libstd/rt.rs:67
  19: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:52
  20: std::panicking::try::do_call
             at src/libstd/panicking.rs:303
  21: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:86
  22: std::panicking::try
             at src/libstd/panicking.rs:281
  23: std::panic::catch_unwind
             at src/libstd/panic.rs:394
  24: std::rt::lang_start_internal
             at src/libstd/rt.rs:51
  25: std::rt::lang_start
             at /usr/src/rustc-1.43.0/src/libstd/rt.rs:67
  26: main
  27: __libc_start_main
  28: _start
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

warning: build failed, waiting for other jobs to finish...
error: failed to run custom build command for `agent v0.0.36 (/home/bburns/src/akri/agent)`

Caused by:
  process didn't exit successfully: `/home/bburns/src/akri/target/debug/build/agent-205b11fae37b158b/build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /home/bburns/.cargo/registry/src/github.com-1ecc6299db9ec823/tonic-build-0.1.1/src/lib.rs:219:19
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /usr/src/rustc-1.43.0/vendor/backtrace/src/backtrace/libunwind.rs:86
   1: backtrace::backtrace::trace_unsynchronized
             at /usr/src/rustc-1.43.0/vendor/backtrace/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:78
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1063
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1426
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:204
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:224
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:474
  11: rust_begin_unwind
             at src/libstd/panicking.rs:378
  12: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
  13: core::result::unwrap_failed
             at src/libcore/result.rs:1222
  14: core::result::Result<T,E>::unwrap
             at /usr/src/rustc-1.43.0/src/libcore/result.rs:1003
  15: tonic_build::fmt
             at /home/bburns/.cargo/registry/src/github.com-1ecc6299db9ec823/tonic-build-0.1.1/src/lib.rs:219
  16: tonic_build::Builder::compile
             at /home/bburns/.cargo/registry/src/github.com-1ecc6299db9ec823/tonic-build-0.1.1/src/lib.rs:172
  17: build_script_build::main
             at agent/build.rs:3
  18: std::rt::lang_start::{{closure}}
             at /usr/src/rustc-1.43.0/src/libstd/rt.rs:67
  19: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:52
  20: std::panicking::try::do_call
             at src/libstd/panicking.rs:303
  21: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:86
  22: std::panicking::try
             at src/libstd/panicking.rs:281
  23: std::panic::catch_unwind
             at src/libstd/panic.rs:394
  24: std::rt::lang_start_internal
             at src/libstd/rt.rs:51
  25: std::rt::lang_start
             at /usr/src/rustc-1.43.0/src/libstd/rt.rs:67
  26: main
  27: __libc_start_main
  28: _start
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by bfjelds** [comment]
@brendandburns, thanks for finding this!

We spun up some ubuntu 20.04 VMs and ran through the build to try to repro this but were unable to.  Can you fill in a few details for us?

* Output of `cat /etc/*-release`
* Where there any errors reported during setup.sh?
* How was rust installed?
* Output of `cargo --version`
* It looks like version 1.43 of rustc is being used, but just to make sure, what is the output of `rustc --version`

**Comment by brendandburns** [comment]
```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04.1 LTS"
NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.1 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

I installed rust w/ `apt-get install cargo` (notably it wasn't installed by the setup script)

Both `cargo` and `rustc` are `1.4.3`

**Comment by brendandburns** [comment]
There was one error in `setup.sh`

```
bburns@helios:~/src/akri$ ./build/setup.sh 
User: bburns
Install rustfmt
./build/setup.sh: line 7: rustup: command not found
```

Maybe that's the issue?

**Comment by brendandburns** [comment]
sorry I didn't notice that error previously as the script kept executing, the script should probably `set -e` at the beginning so that it errors out on errors.

**Comment by brendandburns** [comment]
Sent https://github.com/deislabs/akri/pull/45 for the `set -e`

**Comment by bfjelds** [comment]
There are build dependencies on rustfmt.  I'm not entirely sure how to install it if you use `apt install cargo`, though I will start researching to see if I can find a way.

We have been using rustup (`curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain=1.41.0`) to install Rust.

**Comment by bfjelds** [comment]
You may be able to just run `cargo install rustfmt`?

**Comment by bfjelds** [comment]
It seems that running `cargo install rustfmt` installs an outdated version (0.10.0) that has incompatible arguments.  👎 

Still working on finding a way to get a compatible rustfmt installed with an apt-get installed cargo.

**Comment by bfjelds** [comment]
Rust guidance seems to be that you use `apt remove cargo` before installing and using the rustup toolchain to install cargo, rustc, and rustfmt.

However, I tried using `sudo apt install -y cargo`, **and then** ran the rustup installation command (`curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain=1.41.0` followed by `cargo install rustfmt` ... all of which has now been added to build/setup.sh).  With that, and provided I specify a full path in my cargo build command that matches the rustup installation path (`~/.cargo/bin/cargo build`), the build succeeds for me.

If it were me, I'd probably run `apt remove cargo` and then follow the rustup path (if only because that seems to be the supported Rust path that includes updates and tooling).

Hope this helps!
